AutomatedTesting/Gem/PythonTests/assetpipeline/ap_fixtures/asset_processor_fixture.py

```diff
@@ -36,5 +37,18 @@ def teardown():
         ap.stop()
 
     request.addfinalizer(teardown)
+    assert not check_ap_running("AssetProcessor.exe") and check_ap_running("AssetProcessorBatch.exe"), \
+        "Asset Processor or Asset Processor Batch did not shutdown."
```

Comment:
You might also want to check that there aren't any asset builder executables hanging around? They should always get shut down if AP gets shut down, but tests are good for catching edge cases of things that should happen like that